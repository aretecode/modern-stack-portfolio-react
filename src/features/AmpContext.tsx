/**
 * * @todo need an `AmpContext`
 *   - tweak which tag is used (_can change with `as=`_)
 *   - detect with query in req
 *
 * @todo @@perf https://github.com/ampproject/amp-toolbox/tree/master/packages/optimizer
 */
import { createContext } from 'react'

export interface AmpContextValueType {
  isAmp: boolean
}

export const AmpContext = createContext<AmpContextValueType>({
  isAmp:
    process.env.NODE_ENV !== 'test' && typeof window === 'object'
      ? window.location.href.includes('?amp')
      : false,
})
