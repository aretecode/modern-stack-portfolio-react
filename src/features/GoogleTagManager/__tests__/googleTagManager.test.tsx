import * as React from 'react'
import { render } from '../../../../__tests__/render'
import * as exported from '../index'
import { AmpContext } from '../../AmpContext'
import {
  GoogleTagManagerHeaderScript,
  GoogleTagManagerBodyScript,
} from '../index'

describe('GoogleTagManager', () => {
  it('should export the same stuff', () => {
    expect(exported).toMatchSnapshot()
  })
  it('should render in *non* AMP mode', () => {
    const headScript = render(<GoogleTagManagerHeaderScript />).container
    const bodyScript = render(<GoogleTagManagerBodyScript />).container
    expect(headScript).toMatchSnapshot()
    expect(bodyScript).toMatchSnapshot()
  })
  it('should render *in* AMP mode', () => {
    const headScript = render(
      <AmpContext.Provider value={{ isAmp: true }}>
        <GoogleTagManagerHeaderScript />
      </AmpContext.Provider>
    ).container
    const bodyScript = render(
      <AmpContext.Provider value={{ isAmp: true }}>
        <GoogleTagManagerBodyScript />
      </AmpContext.Provider>
    ).container
    expect(headScript).toMatchSnapshot()
    expect(bodyScript).toMatchSnapshot()
  })
})
