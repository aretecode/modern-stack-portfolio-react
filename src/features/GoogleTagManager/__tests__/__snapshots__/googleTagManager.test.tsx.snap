// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleTagManager should export the same stuff 1`] = `
Object {
  "AnalyticsContainer": [Function],
  "GoogleTagManagerBodyScript": [Function],
  "GoogleTagManagerHeaderScript": [Function],
  "analyticsContainer": AnalyticsContainer {},
  "toGlobalThis": [Function],
  "toGoogleTagManager": [Function],
  "trackEvent": [Function],
}
`;

exports[`GoogleTagManager should render *in* AMP mode 1`] = `
<div>
  <script
    async=""
    custom-element="amp-analytics"
    src="https://cdn.ampproject.org/v0/amp-analytics-0.1.js"
  />
</div>
`;

exports[`GoogleTagManager should render *in* AMP mode 2`] = `
<div>
  <amp-analytics
    config="https://www.googletagmanager.com/amp.json?id=AMP_ID&gtm.url=SOURCE_URL"
    data-credentials="include"
  />
</div>
`;

exports[`GoogleTagManager should render in *non* AMP mode 1`] = `
<div>
  <script>
    
window.dataLayer = []; requestIdleCallback(() =&gt; {
window.dataLayer.push({'gtm.start':new Date().getTime(),event:'gtm.js'});
var f=document.getElementsByTagName('script')[0],j=document.createElement('script');
j.async=true;j.src='https://www.googletagmanager.com/gtm.js?id=WEB_ID';f.parentNode.insertBefore(j,f);
})

  </script>
</div>
`;

exports[`GoogleTagManager should render in *non* AMP mode 2`] = `
<div>
  <noscript />
</div>
`;
